compiler/ballerina-lang/src/main/java/org/ballerinalang/model/types/UnionType.java

```diff
@@ -17,13 +17,13 @@
  */
 package org.ballerinalang.model.types;
 
-import java.util.Set;
+import java.util.LinkedHashSet;
```

Comment:
We need to keep the Set as the type of the variables and use LinkedHashSet only as the implementation.

Reply:
The issue I see there is that if a person subclasses the UnionType, there is no guarantee that it will return an ordered set. Any class implementing the interface Set, even if its not an ordered one, can be used. IMHO we should restrict them to use an ordered set.